rackcity/utils/asset_utils.py

```diff
@@ -456,16 +461,28 @@ def save_all_field_data(data, asset, change_plan=None):
                     and len(assets_with_asset_number) > 0
                     and assets_with_asset_number[0].id != asset.id
                 ):
-                    return (
+                    return (None,
                         "Asset with asset number '"
                         + str(data[field])
                         + "' already exists."
                     )
             value = data[field]
         else:
             value = data[field]
-        setattr(asset, field, value)
-        try:
+
+        print(field, value)
+        if field is not "id":
+            setattr(asset, field, value)
+
+    try:
+        if create_asset_cp:
+            asset_cp = copy_asset_to_new_asset_cp(asset, change_plan)
+            print(asset_cp)
+            asset_cp.save()
+            return asset_cp, None
+
+        else:
             asset.save()
-        except Exception as error:
-            return parse_save_validation_error(error, "Asset")
+            return None, None
+    except Exception as error:
+        return parse_save_validation_error(error, "Asset")
```

Comment:
it looks like you should be returning a tuple everywhere - but a general comment on that - I think it would be more ideal to separate these different paths of logic into separate methods because one of my goals of the first refactor was to make each path more obvious and avoid a lot of if this, do all this, but if not, do all this, etc - if that makes sense? we can talk about this later too

Reply:
cool, i have thoughts on how to fix this but want to run by you before i do it